### PR TITLE
get hotel id feature ticket

### LINKED ISSUE #43: /GET hotels/id
Get a hotel entity given an entity

create the handler and DB accessor, and handler tests for retrieving a single hotel given its id 

Some things to keep in mind: 
- all errors returned from the handler should be the internal error (500) except 400, where the request body is invalid (users from the FE shouldn't need to know what went wrong in our backend), and 404, where it doesn't exists in our DB
- validation logic should be done where necessary

### PR DESCRIPTION
**Pull request by tars919**
## Description

This PR is documenting the implementation for the get hotels id ticket to retrieve a single hotel by its given UUID.

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (code improvement without changing functionality)
- [ ] Documentation update
- [ ] Configuration/infrastructure change
- [ ] Performance improvement
- [ ] Test coverage improvement

## Related Issue(s)

Closes #43 

## What Changed?

- Added the hotel model following the database schema
- Added method getbyid to implement hotelrepository
- Created HotelHandler with UUID validation
- Added tests file to verify functionality
- Registered route

### How to Test

1. Made a test file called hotels_test.go for the handler
2. All the unit tests here passed
3. Verified proper error handling as well

### Test Coverage

- [x] Unit tests added/updated
- [x] Integration tests added/updated
- [x] Manual testing completed
- [x] Edge cases considered and tested

## Unfinished Work & Known Issues

- [x] None, this PR is complete and production-ready
- [ ] The following items are intentionally deferred:
  - ***
  - ***
  - ***

## Notes & Nuances
- First time submitting a PR so not sure if I did this correctly, let me know any issues!

- ***
- ***

## Deployment Considerations

- [x] No special deployment considerations
- [ ] Database migrations required (see `backend/supabase/migrations/`)
- [ ] Environment variables added/changed (document in PR or update `.env.example`)
- [ ] Breaking API changes (requires client updates)

### Code Quality

- [x] Code follows the project's style guidelines and conventions
- [x] Self-review completed (I've reviewed my own code for obvious issues)
- [x] No debugging code, console logs, or commented-out code left behind
- [x] No merge conflicts with the base branch
- [x] Meaningful commit messages that explain the "why"

### Testing & CI

- [x] All CI checks are passing
- [x] All new and existing tests pass locally
- [x] Test coverage hasn't decreased (or decrease is justified)
- [x] Linting passes without errors

### Documentation

- [x] Code is self-documenting or includes helpful comments for complex logic
- [x] API documentation updated (if backend endpoints changed)
- [x] Type definitions are accurate and up-to-date

### Security & Performance

- [x] No sensitive data (passwords, tokens, API keys, PII, etc.) committed
- [x] Security implications considered and addressed
- [x] Performance impact evaluated (no obvious performance regressions)

## Reviewer Notes

**Areas needing extra attention:**

- Questions for reviewers: ...

- Concerns: ...

### CODE REVIEW

File: backend/internal/handler/hotels.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	"github.com/generate/selfserve/internal/errs"
+	"github.com/generate/selfserve/internal/models"
+	
+	"github.com/gofiber/fiber/v2"
+	"github.com/google/uuid"
+)
+
+//Interface for hotel repository -> used in testing later
+type HotelRepository interface {
+	GetByID(ctx context.Context, id string) (*models.Hotel, error)
```
Comment (danctila) [inline backend/internal/handler/hotels.go]: Should be FindByID per naming convention (repository methods for GET should use find prefix). Applies to both the interface and implementation

File: backend/internal/repository/hotels.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+type HotelRepository struct {
+	db *pgxpool.Pool
+}
+
+func NewHotelRepository(db *pgxpool.Pool) *HotelRepository {
+	return &HotelRepository{db: db}
+}
+
+func (r *HotelRepository) GetByID(ctx context.Context, id string) (*models.Hotel, error) {
```
Comment (danctila) [inline backend/internal/repository/hotels.go]: Method should be FindByID per naming convention

**Review by danctila — CHANGES_REQUESTED** [review]
overall great stuff @tars919 🎉. couple of notes and a few minor changes requested mostly to naming.

### good:
- error handling (400, 404, 500) 🙏
- UUID validation is correct, test coverage is epic
- existing repo patterns

### changes requested:
- naming conventions (GetById -> FindByID in repository)
- move `HotelRepository` interface from `storage.go` to `repo_types.go`
- Swagger godoc comments to handler method
